Bottom: 17a47d03c923547e2a42ba9ec06fc6fff21277b7
Top:    5bb552a6d188347d4438003f7bef5d56eb815bce
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-08-07 14:06:31 +0800

macvlan: XDP rx handler support

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/macvlan.c b/drivers/net/macvlan.c
index fed7dda..00e5b56 100644
--- a/drivers/net/macvlan.c
+++ b/drivers/net/macvlan.c
@@ -436,6 +436,16 @@ static void macvlan_forward_source(struct sk_buff *skb,
 	}
 }
 
+int macvlan_handle_frame_xdp(struct xdp_buff *xdp)
+{
+	const struct ethhdr *hdr = (const struct ethhdr *)xdp->data;
+
+	if (is_multicast_ether_addr(eth->h_dest))
+		return -EFAULT;
+
+	
+}
+
 /* called under rcu_read_lock() from netif_receive_skb */
 static rx_handler_result_t macvlan_handle_frame(struct sk_buff **pskb)
 {
